### ISSUE TITLE
LambdaParameterNaming false positives on data class constructors

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5433]
## Expected Behavior
No findings.

## Observed Behavior
```kotlin
data class UserInfoOpenID(
	...
	/**
	 * The user's given name(s) or first name(s).
	 * _Might be provided when a [name] claim is present._
	 * @since profile
	 */
	val given_name: String?,
```
is flagged as:
```
LambdaParameterNaming - [Lambda parameter names should match the pattern: [a-z][A-Za-z0-9]*|_] at Heroku\backend\endpoint\src\main\kotlin\net\twisterrob\cinema\cineworld\backend\endpoint\auth\data\UserInfoOpenID.kt:35:6
```

## Steps to Reproduce
```kotlin
    @Test
    fun `doesn't report invalid class constructor parameters`() {
        val code = """
            data class C(val _invalid: String)
        """.trimIndent()
        assertThat(LambdaParameterNaming().compileAndLint(code))
            .isEmpty()
    }

    @Test
    fun `doesn't report invalid function parameters`() {
        val code = """
            fun f(val _invalid: String) = Unit
        """.trimIndent()
        assertThat(LambdaParameterNaming().compileAndLint(code))
            .isEmpty()
    }
```

## Context
Could be caused by https://github.com/detekt/detekt/pull/5212 cc @VovaStelmashchuk @cortinico.

In the original code it was like this (notice the override vs the delegated method):
```kotlin
    override fun visitLambdaExpression(lambdaExpression: KtLambdaExpression) {
        super.visitLambdaExpression(lambdaExpression)
        lambdaExpression.valueParameters
            .forEach { lambdaParameterNamingRule.runIfActive { visitParameter(it) } }
    }
```

## Your Environment

* Version of detekt used: 1.22.0-RC1 and 1.22.0-RC2 (1.21.0 was fine)
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.cinema/tree/356d7629cf1191fd1ebba7ad094f09b0cef307f8/Heroku
